next.config.js

```diff
@@ -107,49 +128,80 @@ const nextConfig = {
                     },
                 ],
             },
+            {
+                source: '/sw.js',
+                headers: [
+                    {
+                        key: 'Content-Type',
+                        value: 'application/javascript; charset=utf-8',
+                    },
+                    {
+                        key: 'Cache-Control',
+                        value: 'no-cache, no-store, must-revalidate',
+                    },
+                    {
+                        key: 'Content-Security-Policy',
+                        value: "default-src 'self'; script-src 'self'; connect-src 'self' https:; worker-src 'self'",
+                    },
+                ],
+            },
         ]
     },
 }
 
-const { withSentryConfig } = require('@sentry/nextjs')
+module.exports = async (phase) => {
+    let conf = nextConfig
+    if (phase === PHASE_DEVELOPMENT_SERVER || phase === PHASE_PRODUCTION_BUILD) {
+        console.log('withSerwist')
+        const withSerwist = (await import('@serwist/next')).default({
+            swSrc: 'src/app/sw.ts',
+            swDest: 'public/sw.js',
+        })
+
+        conf = withSerwist(conf)
+    }
```

Comment:
Q: what does this do :sob:

Reply:
This is the service worker utility, it basically compiles your sw.ts file to javascript into the /public directory for runtime usage

Reply:
Also this module can't be required at the top and need to be imported dynamically